File: app/src/main/java/com/FaceLab/FaceLab/TestActivity.java
package com.FaceLab.FaceLab;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.PorterDuff;
import android.graphics.PorterDuffXfermode;
import android.graphics.Rect;
import android.graphics.RectF;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;

import com.google.android.gms.ads.AdListener;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdView;
import com.google.android.gms.ads.MobileAds;

public class TestActivity extends AppCompatActivity {
    private byte[] byteArray;
    private Bitmap bmp;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_test);
        MobileAds.initialize(this, getString(R.string.admob_app_id));

        ImageView ivFace = findViewById(R.id.test_iv_face);
        AdView adView = findViewById(R.id.adView);
        AdRequest adRequest = new AdRequest.Builder().build();
        adView.loadAd(adRequest);

        // 광고가 제대로 로드 되는지 테스트 하기 위한 코드입니다.
        adView.setAdListener(new AdListener() {
            @Override
            public void onAdLoaded() {
                // Code to be executed when an ad finishes loading.
                // 광고가 문제 없이 로드시 출력됩니다.
                Log.d("@@@", "onAdLoaded");
            }
            @Override
            public void onAdFailedToLoad(int errorCode) {
                // Code to be executed when an ad request fails.
                // 광고 로드에 문제가 있을시 출력됩니다.
                Log.d("@@@", "onAdFailedToLoad " + errorCode);
            }
            @Override
            public void onAdOpened() {
                // Code to be executed when an ad opens an overlay that
                // covers the screen.
            }
            @Override
            public void onAdClicked() {
                // Code to be executed when the user clicks on an ad.
            }
            @Override
            public void onAdLeftApplication() {
                // Code to be executed when the user has left the app.
            }
            @Override
            public void onAdClosed() {
                // Code to be executed when the user is about to return
                // to the app after tapping on an ad.
            }
        });

        byteArray = getIntent().getByteArrayExtra("image");
        assert byteArray != null;
        bmp = BitmapFactory.decodeByteArray(byteArray, 0, byteArray.length);
        ivFace.setImageBitmap(getRoundedCornerBitmap(bmp));
    }

    public void customOnClick(View v) {
        if (v.getId() == R.id.test_cl_analysis) {
            TestModel model = new TestModel(this, bmp);
            model.runModel();

            Intent intent = new Intent(this, ResultActivity.class);
            intent.putExtra("image", byteArray);
            intent.putExtra("output", model.getOutput());
            intent.putExtra("sexAge", model.getAgeSex());
            intent.putExtra("emotion", model.getEmotion());
            intent.putExtra("result", model.getResult());
            startActivity(intent);
        }
    }

    //모서리가 둥근 이미지를 보여주는 함수
    public static Bitmap getRoundedCornerBitmap(Bitmap bitmap) {

        Bitmap output = Bitmap.createBitmap(bitmap.getWidth(), bitmap
                .getHeight(), Bitmap.Config.ARGB_8888);
        Canvas canvas = new Canvas(output);
        final int color = 0xff424242;
        final Paint paint = new Paint();
        final Rect rect = new Rect(0, 0, bitmap.getWidth(), bitmap.getHeight());
        final RectF rectF = new RectF(rect);
        final float roundPx = 40;

        paint.setAntiAlias(true);
        canvas.drawARGB(0, 0, 0, 0);
        paint.setColor(color);
        canvas.drawRoundRect(rectF, roundPx, roundPx, paint);

        paint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.SRC_IN));
        canvas.drawBitmap(bitmap, rect, rect, paint);

        return output;
    }
}

File: app/src/main/java/com/FaceLab/FaceLab/ui/TestActivity.kt
package com.FaceLab.FaceLab.ui

import android.content.Intent
import android.graphics.*
import android.net.Uri
import android.os.Bundle
import android.provider.MediaStore
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.FaceLab.FaceLab.databinding.ActivityTestBinding
import com.FaceLab.FaceLab.model.TestModel
import com.google.android.gms.ads.AdListener
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.MobileAds
import java.io.IOException

class TestActivity : AppCompatActivity() {
    private var imageUri: Uri? = null
    private var byteArray: ByteArray? = null
    private var bmp: Bitmap? = null
    private lateinit var binding: ActivityTestBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityTestBinding.inflate(layoutInflater)
        setContentView(binding.root)

        MobileAds.initialize(this) { }
        val adRequest = AdRequest.Builder().build()
        binding.adView.loadAd(adRequest)

        // 광고가 제대로 로드 되는지 테스트 하기 위한 코드입니다.
        binding.adView.adListener = object : AdListener() {
            override fun onAdLoaded() {
                // Code to be executed when an ad finishes loading.
                // 광고가 문제 없이 로드시 출력됩니다.
                Log.d("@@@", "onAdLoaded")
            }

            override fun onAdOpened() {
                // Code to be executed when an ad opens an overlay that
                // covers the screen.
            }

            override fun onAdClicked() {
                // Code to be executed when the user clicks on an ad.
            }

            override fun onAdClosed() {
                // Code to be executed when the user is about to return
                // to the app after tapping on an ad.
            }
        }

        imageUri = Uri.parse(intent.extras!!.getString("imageUri"))
        try {
            bmp = MediaStore.Images.Media.getBitmap(contentResolver, imageUri)
            binding.testIvFace.setImageBitmap(bmp)
        } catch (e: IOException) {
            e.printStackTrace()
        }

        binding.testClAnalysis.setOnClickListener { startResultActivity() }
    }

    fun startResultActivity() {
        val model = TestModel(this, bmp!!)
        model.runModel()
        val intent = Intent(this, ResultActivity::class.java)
        intent.putExtra("imageUri", imageUri.toString())
        intent.putExtra("image", byteArray)
        intent.putExtra("output", model.output)
        intent.putExtra("sexAge", model.ageSex)
        intent.putExtra("emotion", model.emotion)
        intent.putExtra("result", model.result)
        startActivity(intent)
        finish()
    }
}